integration-tests/src/tests/features/in_memory_tries.rs

```diff
@@ -419,31 +416,24 @@ fn test_in_memory_trie_consistency_with_state_sync_base_case(track_all_shards: b
 
     let mut clock = FakeClock::new(Utc::UNIX_EPOCH);
 
-    let epoch_length = 10;
     let shard_layout = ShardLayout::simple_v1(&["account3", "account5", "account7"]);
     let validators_spec = ValidatorsSpec::desired_roles(
         &accounts[0..NUM_VALIDATORS].iter().map(|a| a.as_str()).collect::<Vec<_>>(),
         &[],
     );
 
-    let (genesis, epoch_config_store) = build_genesis_and_epoch_config_store(
-        GenesisAndEpochConfigParams {
-            epoch_length,
-            protocol_version: PROTOCOL_VERSION,
-            shard_layout,
-            validators_spec,
-            accounts: &accounts,
-        },
-        |genesis_builder| {
-            genesis_builder
-                .genesis_time_from_clock(&clock.clock())
-                .genesis_height(10000)
-                .transaction_validity_period(1000)
-        },
-        |epoch_config_builder| {
-            epoch_config_builder.minimum_validators_per_shard(NUM_VALIDATORS_PER_SHARD as u64)
-        },
-    );
+    let genesis = TestGenesisBuilder::new()
+        .genesis_time_from_clock(&clock.clock())
+        .genesis_height(10000)
+        .epoch_length(10)
+        .transaction_validity_period(1000)
+        .shard_layout(shard_layout)
+        .validators_spec(validators_spec)
+        .add_user_accounts_simple(&accounts, initial_balance)
+        .build();
+    let epoch_config_store = TestEpochConfigBuilder::from_genesis(&genesis)
+        .minimum_validators_per_shard(NUM_VALIDATORS_PER_SHARD as u64)
+        .build_store_for_single_version(genesis.config.protocol_version);
```

Comment:
As an optional suggestion, it would be nice if we could skip having to provide the protocol version to `build_store_for_single_version()`. I don't see a case where we would need to provide anything other than `genesis.config.protocol_version`.

To achieve this, we store the `genesis_protocol_version: Option<ProtocolVersion>` in `TestEpochConfigBuilder` and populate it during call to `TestEpochConfigBuilder::from_genesis(&genesis)`.